File: strategy/php/mallardduck.php
<?php
/**
 * File mallardduck.php
 *
 * @author kujaomega
 */

/**
 * This is the class for mallardDuck
 *
 * class MallardDuck
 */
class MallardDuck extends Duck
{
    /**
     * This is the constructor
     */
    public function __construct()
    {
        $this->quack_behavior = new Quack();
        $this->fly_behavior = new FlyWithWings();
    }

    /**
     * This is the display method.
     */
    public function display()
    {
        echo( 'I\'m a real Mallard Duck' );
    }
}

File: strategy/java/MallardDuck.java
/*
 * File MallardDuck.java
 * 
 * @author kujaomega
 */

/**
 * This class will be a mallard duck.
 * 
 * class MallardDuck
 */
public class MallardDuck extends Duck{
    
    /**
     * This is the constructor of the mallard duck class.
     */
    public MallardDuck(){
        quackBehavior = new Quack();
        flyBehavior=new FlyWithWings();
        
    }
    
    /**
     * This is the display method of the mallard duck.
     */
    public void display(){
        System.out.println("I'm a real Mallard Duck");
    }
}
